go/def.bzl

```diff
@@ -431,6 +431,17 @@ def _emit_go_link_action(ctx, importmap, transitive_libs, cgo_deps, lib,
     if d.basename.endswith('.so'):
       dirname = _short_path(d)[:-len(d.basename)]
       ldflags += ["-Wl,-rpath,$ORIGIN/" + ("../" * pkg_depth) + dirname]
+  filtered_gc_linkopts = []
+  is_extldflags = False
```

Comment:
Good idea. I've factored out _extract_extldflags, which returns an updated version of both lists.

Looks like something's going wrong with the Mac build; will fix on my laptop when I get home today.